### PR TITLE
Internal: Added elementor regression template test [ED-10233]

### PR DESCRIPTION
**Pull request by dmaziar**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: new playwright template test.

## Summary

This PR can be summarized in the following changelog entry:
1. Added template tests with three templates.
2. Fixed tooltip problem, that fails element-regression tests.
3. Added parallel run for element-regression tests with three workers.
4. Converted modules to ECMA 6 standard.
*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/elements-regression/src/pages/editor-page.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
-const BasePage = require( './base-page.js' );
+export default class EditorPage {
```
Comment (yotamselementor) [inline tests/elements-regression/src/pages/editor-page.js:1]: Can we use the same EditorPage that we use in the sanity tests?

File: tests/elements-regression/src/controls/choose.js
Comment (yotamselementor) [inline tests/elements-regression/src/controls/choose.js:1]: Do we need these controls with the new templates tests?

File: tests/elements-regression/src/pages/wp-admin-page.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
-const BasePage = require( './base-page.js' );
+export default class WpAdminPage {
```
Comment (yotamselementor) [inline tests/elements-regression/src/pages/wp-admin-page.js:1]: Can we reuse the sanity tests' WPAdminPage?

File: tests/elements-regression/test.config.js
Comment (yotamselementor) [inline tests/elements-regression/test.config.js:1]: Do we still need this file?

File: tests/elements-regression/tests/regression_with_templates.test.js
Comment (yotamselementor) [inline tests/elements-regression/tests/regression_with_templates.test.js:1]: This file can be the new implementation of `tests/elements-regression/tests/elements-regression.test.js` -> Can this just replace the content of `tests/elements-regression/tests/elements-regression.test.js`?

File: tests/elements-regression/tests/regression_with_templates.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+			await editorPage.loadTemplate( filePath );
+			await editorPage.setElementorEditorPanel( 'Preview' );
+			const widgetCount = await editorPage.getWidgetCount();
+			for ( let i = 0; i < widgetCount; i++ ) {
+				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
+			}
+			await editorPage.setElementorEditorPanel( 'Back to Editor' );
+			await editorPage.publishAndViewPage();
+			await page.waitForTimeout( 500 );
+			await expect( editorPage.container ).toHaveScreenshot( `${ widgetType }_published.png`, { maxDiffPixels: 100 } );
```
Comment (yotamselementor) [inline tests/elements-regression/tests/regression_with_templates.test.js:33]: Can we do something similar to what you did here?
```typescript
			for ( let i = 0; i < widgetCount; i++ ) {
				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
			}
```

File: tests/elements-regression/tests/regression_with_templates.test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			await editorPage.setupElementorPage( wpAdminPage );
+			await editorPage.loadTemplate( filePath );
+			await editorPage.setElementorEditorPanel( 'Preview' );
+			const widgetCount = await editorPage.getWidgetCount();
+			for ( let i = 0; i < widgetCount; i++ ) {
+				await expect( editorPage.getWidget().nth( i ) ).toHaveScreenshot( `${ widgetType }_${ i }.png` );
+			}
+			await editorPage.setElementorEditorPanel( 'Back to Editor' );
+			await editorPage.publishAndViewPage();
+			await page.waitForTimeout( 500 );
```
Comment (yotamselementor) [inline tests/elements-regression/tests/regression_with_templates.test.js:32]: Do we need this?

File: tests/elements-regression/tests/templates/text_editor.json [hunk 5]
```diff
@@ -0,0 +1,108 @@
+{
```
Comment (yotamselementor) [inline tests/elements-regression/tests/templates/text_editor.json:1]: You are missing some of the permutations that we had in the original tests
Also, there are missing templates for `image` and `button`

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: tests/elements-regression/src/controls/choose.js
Comment (TzviRabinovitch) [reply tests/elements-regression/src/controls/choose.js:1]: @dmaziar 
maybe move to a location that we can reuse in other playwright tests ?

Comment (dmaziar) [reply tests/elements-regression/src/controls/choose.js:1]: @TzviRabinovitch @yotamselementor should I remove old solution and leave only with templates?

Comment (yotamselementor) [reply tests/elements-regression/src/controls/choose.js:1]: @dmaziar yes, lets just keep the new one plz

File: tests/elements-regression/src/pages/editor-page.js (hunk 1 above)
Comment (dmaziar) [reply tests/elements-regression/src/pages/editor-page.js]: fixed

File: tests/elements-regression/src/controls/choose.js
Comment (dmaziar) [reply tests/elements-regression/src/controls/choose.js:1]: Old element regression is removed.  The code that could be reused is moved to other pages.

File: tests/elements-regression/tests/regression_with_templates.test.js
Comment (dmaziar) [reply tests/elements-regression/tests/regression_with_templates.test.js:1]: fixed.

File: tests/elements-regression/tests/regression_with_templates.test.js (hunk 3 above)
Comment (dmaziar) [reply tests/elements-regression/tests/regression_with_templates.test.js]: As we discussed in slack, before publish it makes sense to get screenshot by widget. But after publish we can check everything with just one. The important thing is to do it in one container

File: tests/elements-regression/tests/regression_with_templates.test.js (hunk 4 above)
Comment (dmaziar) [reply tests/elements-regression/tests/regression_with_templates.test.js]: removed from here, but before publish we have to use explicit timeout. I didn't found a solution of how to wait until all widgets are rendered. It's hard to do it in Iframe. Event this didn't help: `await editorPage.waitForElementRender( widgetIds[ 0 ] );`

File: tests/elements-regression/tests/templates/text_editor.json (hunk 5 above)
Comment (dmaziar) [reply tests/elements-regression/tests/templates/text_editor.json:1]: fixed. Also added some description of what is being tested in templates.

**Review by yotamselementor — APPROVED** [review]

**Pull request merged**